### ISSUE TITLE
Enhancement: ability to clean up recursive trace spam

### ISSUE THREAD

**Issue by tgschultz** [ziglang/zig#2386]
Opening issue per discussion in IRC.

In my case I recursively generated a type (see: [pseudo-struct](https://gist.github.com/tgschultz/e168cffd69003e4c788b6fd2ab1aea9e)) where a compile error was thrown if the 'bottom' of the stack was reached without a being able to fulfill the request. This produced the following error trace:

```
...\object_store.zig:557:21: error: u9 not a member of this MultiObjectPool
                    @compileError(@typeName(O) ++ " not a member of this MultiObjectPool");
                    ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:659:28: note: called from here
        mop.trimEmptyStores(u9, .Smallest);
                           ^
```

As you can see, most of these entries are pointing to the same thing. In known recursive cases, it would be nice to have the ability to condense the trace so it looks more like this:
```
...\object_store.zig:557:21: error: u9 not a member of this MultiObjectPool
                    @compileError(@typeName(O) ++ " not a member of this MultiObjectPool");
                    ^

...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
...\object_store.zig:659:28: note: called from here
        mop.trimEmptyStores(u9, .Smallest);
                           ^
```

**Comment by tgschultz** [comment]
As noted in IRC, one potential way to handle this is using an output filter that checks if the current message is identical to the last message, keeping a count, then outputting something like `(repeated N times)`. Example:
```
...\object_store.zig:557:21: error: u9 not a member of this MultiObjectPool
                    @compileError(@typeName(O) ++ " not a member of this MultiObjectPool");
                    ^

...\object_store.zig:556:70: note: called from here
                    if(Prev != void) return self.prev.trimEmptyStores(O, keep_type);
                                                                     ^
(repeated 9 times)

...\object_store.zig:659:28: note: called from here
        mop.trimEmptyStores(u9, .Smallest);
                           ^
```

**Comment by andrewrk** [comment]
This has been solved with [Referenced-By Traces](https://ziglang.org/download/0.10.0/release-notes.html#Referenced-By-Traces) which landed in 0.10.0.
